paddle/fluid/pybind/imperative.cc

```diff
@@ -469,6 +469,39 @@ static void ParseIndexingSlice(framework::LoDTensor *tensor, PyObject *_index,
   if (!PyTuple_Check(_index)) Py_DecRef(index);
 }
 
+template <typename P>
+static void VarBaseCopy(const imperative::VarBase &src,
+                        imperative::VarBase &dst, const P &dst_device,
+                        const bool blocking) {
+  if (dst.SharedVar()->IsEmpty()) {
+    VLOG(3) << "deep copy Variable from " << src.Name() << " to " << dst.Name();
+    dst.SetPersistable(src.Persistable());
+    dst.SetDataType(src.DataType());
+    dst.SetType(src.Type());
+    dst.SetOverridedStopGradient(src.OverridedStopGradient());
+    if (!src.SharedVar()->IsEmpty()) {
```

Comment:
same above, add else branch and throw error to avoid bug, can use `PADDLE_THROW`